### PR TITLE
doc: update versions for 1.4.0 release

### PR DESCRIPTION
**Pull request by greg-fer**
Updated version files and version mentions for sdk-nrf.
Updated SES link to point to v510c, and removed unused and duplicated links.
Updated manifest.
NCSDK-5798

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

- [x] Update SES links to 5.10d after #3137 is merged.
- [x] Update west.yml

### CODE REVIEW

File: README.rst
Comment (ru-fu) [inline README.rst:28]: No - last take on this was to not replace the version, because it won't be resolved in GitHub.
See https://github.com/nrfconnect/sdk-nrf/pull/3082

Comment (greg-fer) [reply README.rst]: Ok, thanks.

**Review by tejlmand — APPROVED** [review]

**Comment by greg-fer** [comment]
Needs an update of SES links after #3137 goes in.

**Review by wbober — CHANGES_REQUESTED** [review]
@greg-fer I might need one subtle change. I'll confirm tomorow.

**Review by wbober — APPROVED** [review]

**Review by b-gent — APPROVED** [review]
Approving. Please remember https://github.com/nrfconnect/sdk-nrfxlib/pull/300 must go in first, and then update west.yml here before merging.

File: doc/nrfxlib/conf.py
Comment (ru-fu) [inline doc/nrfxlib/conf.py:95]: This change is missing for nrf/conf.py

**Review by ru-fu — CHANGES_REQUESTED** [review]

File: doc/versions.json
Comment (carlescufi) [inline doc/versions.json:19]: Why is this 2.4.99? This should be 2.4.0

**Review by carlescufi — CHANGES_REQUESTED** [review]

File: doc/versions.json
Comment (greg-fer) [reply doc/versions.json]: Fixed

**Review by carlescufi — APPROVED** [review]

File: doc/versions.json
Comment (ru-fu) [reply doc/versions.json]: Actually, 2.4.99 is more correct, since this is what will be displayed for the /latest/ folder. In the /v1.4.0/ folder, the data from the v1.4.0 entry is displayed.
It doesn't really make a difference if it is changed in this PR or in https://github.com/nrfconnect/sdk-nrf/pull/3240 though.

**Comment by greg-fer** [comment]
Do not merge until tomorrow morning, Friday Oct 30.

**Review by ru-fu — APPROVED** [review]

**Pull request merged**
